src/globus_cli/termio/output_formatter.py

```diff
@@ -282,6 +284,17 @@ def _print_as_text():
         elif text_format == FORMAT_TEXT_RECORD:
             _assert_fields()
             colon_formatted_print(data, fields)
+        elif text_format == FORMAT_TEXT_RECORD_LIST:
+            _assert_fields()
+            if not isinstance(data, list):
+                raise ValueError("only lists can be output in text record list format")
```

Comment:
The requirement for `list` is the only thing that makes me hesitate here. If we were to pass a paginated call, it would be cool to be able to stream output.

Maybe it's better to check `collections.abc.Iterable`?

We can leave this as-is until we have a use-case for a non-list iterable.

Reply:
Agh, yeah. The thing that makes me hesitate with Iterable (or trying `iter()`) is that it allows dicts, which I think is against the intent here. I guess `isinstance(Iterable) and not isinstance(dict)`?

Reply:
What do you think of this rendition?

Reply:
To be straight with you, I prefer the original.
When it's done via `iter(x)` with the try-catch, it's a lot of cognitive/reading overhead for this. Is `isinstance(x, collections.abc.Iterable)` not viable? Have I forgotten that being missing in py3.6 or something?

There are other mapping types and mapping-likes, `GlobusHTTPResponse` in particular, which can get passed. If we're not going to accept an arbitrary iterable, I'd probably just stick with `list` for today and we can always revisit and expand it later.

Reply:
I vaguely recall reading some encouragement to use the try-iter version since it was more reliable than `Iterable`, but that's probably outdated. I think I'm operating on superstition here.

I'm realizing also `collections.abc.Sequence` is very close to what we want, but that would require also adding a `__len__` to `IterableResponse`. I'm good to just assert list here for now and revisit.

Reply:
FWIW, you're not alone. I've definitely done things in the past because it's what we had to do for py2.6 compatibility, and I never "re-learned" how to do it. With the rate of changes in typing, I suspect that for some time we'll be (re)learning how we're "supposed" to express iterable, comparable, mapping, frozen, truthy/falsey, and a wide array of other concepts that loosely fit `typing.Protocol`.